scripts/workspace-conversion/README.md

```diff
@@ -0,0 +1,60 @@
+## Converting Workspace Dataset Attributes to Snapshot Data
+
+# Description
+This script takes the workspace dataset attributes and attempts to map them to similar fields in the data catalog service as snapshots.
+
+# Dataset Attributes
+An example workspace with attributes can be found [here](src/data/workspace-cleanAttributes.js.displayLibraryAttributes).
+
+The request made to retrieve the workspace details is an XHR request to rawls to retrieve the workspace information. The basic structure of the return of the call looks like:
+
+```
+{
+  owners: [],
+  workspaces: {
+    attributes: {
+      description: string,
+      `library:cellType: string,
+      `library: ...
+    }
+  }
+}
+```
+
+What is shown to the user in that table is the contents in the attributes that terra-ui has determined is important to display. These values are hardcoded at 
+`{terra-ui}/src/data/workspace-cleanAttributes.js.displayLibraryAttributes`
+
+# Value mappings
+To run the conversion from workspace to dataset, here is a current mapping scheme:
+
+| Attribute | Rawls Workspace Path | Data Catalog Snapshot Path |
+| --------- | -------------------- | -------------------------- |
+| Cohort Phenotype/Indication (Disease Ontology)  | `library:diseaseOntologyLabel` |  |
+| No. of Subjects | `library:numSubjects` | `counts.donors` |
+| Data Category | `library:dataCategory`<br>ex: "Simple Nucleotide Variation, Copy Number Variation, Expression Quantification, DNA-Methylation, Clinical phenotypes, Biosample metadata" |  |
+| Experimental Strategy | `library:datatype`<br>ex: "Whole Exome, Genotyping Array, RNA-Seq, miRNA-Seq, Methylation Array, Protein Expression Array" | `prov:wasGeneratedBy.3.TerraCore:hasAssayCategory`<br>Note: RNA-Seq would be found in this field in existing data catalog entries, but I don't know if all of these values would also fit in the same one. |
+| Data Use Limitation | `library:dataUseRestriction` | `TerraDCAT_ap:hasDataUsePermission.0`<br>Not a perfect mapping, we will need to normalize "General Research Use" to "TerraCore:NoRestriction" |
+| Cohort Phenotype/Indication | `library:indication`<br>ex: "Mesothelioma" | `samples.disease.0` |
+| Cohort Name | `library:datasetName` | `dct:title` |
+| Dataset Version | `library:datasetVersion` |  |
+| Cohort Description | `library:datasetDescription` | `dct:description` |
+| Dataset Owner | `library:datasetOwner`<br>ex: "NCI" | `TerraDCAT_ap:hasDataCollection.0.dct:identifier`<br>If we have enough information, we can also build:<br> `TerraDCAT_AP:hasDataCollection.0.dct:publisher: National Cancer Institute`<br>`TerraDCAT_AP:hasDataCollection.0.dct:title: National Cancer Institute` |
+| Dataset Custodian | `library:datasetCustodian`<br>ex: "dbGAP" | `TerraDCAT_ap:hasDataCollection.0.dct:identifier`<br>Note: This conflicts with "Dataset Owner", but I dont think we have another option for where to put this. |
```

Comment:
I think the TIM seperates the owner from the publisher, so we might be able to use that.